sdks/python/apache_beam/runners/portability/fn_api_runner_transforms.py

```diff
@@ -788,14 +799,34 @@ def make_stage(base_stage, transform_id, extra_must_follow=()):
             inputs=dict(transform.inputs, **{main_input_tag: paired_pcoll_id}),
             outputs={'out': split_pcoll_id})
 
+        if common_urns.composites.RESHUFFLE.urn in context.known_runner_urns:
+          reshuffle_pcoll_id = copy_like(
+              context.components.pcollections,
+              main_input_id,
+              '_reshuffle',
+              coder_id=sized_coder_id)
+          reshuffle_transform_id = copy_like(
+              context.components.transforms,
+              transform,
+              unique_name=transform.unique_name + '/Reshuffle',
+              urn=common_urns.composites.RESHUFFLE.urn,
+              payload=b'',
+              inputs=dict(transform.inputs, **{main_input_tag: split_pcoll_id}),
+              outputs={'out': reshuffle_pcoll_id})
+          yield make_stage(stage, reshuffle_transform_id)
+        else:
+          reshuffle_pcoll_id = split_pcoll_id
+          reshuffle_transform_id = None
```

Comment:
Isn't Reshuffle necessary to balance split work evenly?

Reply:
Some runners have the notion of fusion breaks (see the mustfollow below) that can do this. The direct/in-process runners don't care.